mycroft/skills/core.py

```diff
@@ -699,9 +711,45 @@ def cancel_event(self, name):
             Args:
                 name (str):   Name of event
         """
-        data = {'event': self._unique_name(name)}
+        unique_name = self._unique_name(name)
+        data = {'event': unique_name}
+        self.remove_event(unique_name)
         self.emitter.emit(Message('mycroft.scheduler.remove_event', data=data))
 
+    def get_event_status(self, name):
+        """
+            Get event data and return the amount of time left
+
+            Args:
+                name (str): Name of event
+
+            Return:
+                time_left (int): the time left in seconds
+        """
+        event_name = self._unique_name(name)
+        data = {'name': event_name}
+
+        # making event_status an object so it's refrence can be changed
+        event_status = [None]
+
+        def callback(message, event_status):
+            event_time = int(message.data[0][0])
```

Comment:
Needs check if `message.data is not None` since that is a valid response if event doesn't exist. If it is `None` set event_status[0] to False or 0 to indicate that it doesn't exist.

(Timeout exception is a bit misleading if the actual issue is that the handler doesn't exist)